pulsar4s-cats-effect/src/main/scala/com/sksamuel/pulsar4s/cats/CatsAsyncHandler.scala

```diff
@@ -3,66 +3,75 @@ package com.sksamuel.pulsar4s.cats
 import java.util.concurrent.CompletableFuture
 import java.util.function.BiConsumer
 
-import cats.effect.IO
+import cats.implicits._
+import cats.effect._
 import com.sksamuel.pulsar4s.{AsyncHandler, ConsumerMessage, MessageId}
 import org.apache.pulsar.client.api
 import org.apache.pulsar.client.api.{Reader, TypedMessageBuilder}
 
-import scala.language.implicitConversions
+import scala.language.higherKinds
 import scala.util.{Failure, Success, Try}
 
-class CatsAsyncHandler extends AsyncHandler[IO] {
 
-  implicit def completableVoidToIO(f: => CompletableFuture[Void]): IO[Unit] = completableToIO(f).map(_ => ())
-  implicit def completableToIO[T](f: => CompletableFuture[T]): IO[T] =
-    IO.async[T] { k =>
-      f.whenCompleteAsync(new BiConsumer[T, Throwable] {
-        override def accept(t: T, e: Throwable): Unit = {
-          if (e != null) k.apply(Left(e))
-          else k.apply(Right(t))
+object CatsAsyncHandler {
+  object CompletableFutureConverters {
+    implicit class CompletableOps[T](f: => CompletableFuture[T]) {
+      def toF[F[_]: Async]: F[T] =
+        Async[F].async[T] { k =>
+          f.whenCompleteAsync(new BiConsumer[T, Throwable] {
+            override def accept(t: T, e: Throwable): Unit = {
+              if (e != null) k.apply(Left(e))
+              else k.apply(Right(t))
+            }
+          })
         }
-      })
+
     }
+  }
 
-  override def failed(e: Throwable): IO[Nothing] = IO.raiseError(e)
+  implicit def asyncHandlerForCatsEffectAsync[F[_]: Async]: AsyncHandler[F] = new AsyncHandler[F] {
```

Comment:
Does this need to be `implicit`? It seems like it's always being explicitly called.

Reply:
this needs to be implicit to provide `AsyncHandler` instances for any `F[_]` that has an `Async`, you can see it's usage in the new tests I've added